### PR TITLE
Integrate Message class to manual_submission

### PR DESCRIPTION
**Pull request by JEThompson97**
### Summary of work
- This PR begins the task of Integrating the Message class into the code base (#512)
- Specifically, this PR does the following:
  * Updates `QueueClient.send` to accommodate Message instances being received as the message argument (as well as json dumps which have been use up until now)
  * Adds logging to the Message class such that when `populate` encountered a key which does not match an attribute, a warning is logged
  * Refactors the `manual_submission` call to `QueueClient.send` to use the Message class instead of a json dump (+ tests)
  * Adds a test for a usage of `QueueClient.send` that was missed.
    * note: there is another usage in the same class but it is within the 100-line `data-ready` method.  Due to it's size, adding tests to this method should be separate chunk of work.

### How to test your work
- Ensure all tests pass
- Code review

Part of #512

**Before merging ensure the release notes have been updated**

### CODE REVIEW

File: message/job.py [hunk 1]
```diff
@@ -35,12 +42,14 @@ class Message:
     return_message = attr.ib(default=None)
     retry_in = attr.ib(default=None)
 
-    def serialize(self):
+    # Note: added indent to allow pretty-printing
```
Comment (ElliotAOram) [inline message/job.py:45]: if defaults to None this is fine, but just be sure to add it in the param docstring and give explanation

File: queue_processors/queue_processor/tests/test_queue_processor.py [hunk 2]
```diff
@@ -42,3 +44,27 @@ def test_setup_connection(self, mock_sub_ar, mock_connect, mock_client):
...
+
+class TestListener(unittest.TestCase):
+    """
+    Exercises the Listener
+    """
+    def setUp(self):
+        self.rb_number = -1
+        self.reason = "test"
+
+    def test_construct_and_send_skipped(self):
```
Comment (ElliotAOram) [inline queue_processors/queue_processor/tests/test_queue_processor.py:57]: 👍

File: utils/clients/tests/test_queue_client.py [hunk 3]
```diff
@@ -102,16 +103,29 @@ def test_serialise_message(self):
 
     # pylint:disable=no-self-use
     @patch('stomp.connect.StompConnection11.send')
```
Comment (ElliotAOram) [inline utils/clients/tests/test_queue_client.py:105]: Can we change this to mock `utils.clients.queue_client.QueueClient.send`

File: utils/clients/tests/test_queue_client.py [hunk 4]
```diff
@@ -102,16 +103,29 @@ def test_serialise_message(self):
...
         """
         client = QueueClient()
-        client.send('dataready', 'test-message')
+        client.send('dataready', 'raw_json_dump')
         (args, _) = mock_stomp_send.call_args
         self.assertEqual(args[0], 'dataready')
-        self.assertEqual(args[1], 'test-message')
+        self.assertEqual(args[1], 'raw_json_dump')
+
+    @patch('stomp.connect.StompConnection11.send')
```
Comment (ElliotAOram) [inline utils/clients/tests/test_queue_client.py:117]: same as above re-patch statement

Comment (ElliotAOram) [reply utils/clients/tests/test_queue_client.py:117]: It should be a like for like conversion (code shouldn't have to change other than what it is patching

File: queue_processors/queue_processor/tests/test_queue_processor.py [hunk 5]
```diff
@@ -42,3 +44,27 @@ def test_setup_connection(self, mock_sub_ar, mock_connect, mock_client):
...
+        # pylint:disable=protected-access
+        """
+        Test: _data_dict['message'] is given a value,and the _data_dict is sent via the QueueClient
```
Comment (JEThompson97) [inline queue_processors/queue_processor/tests/test_queue_processor.py:60]: Note to self: add space
> ..value, and..

File: utils/clients/tests/test_queue_client.py (hunk 4 above)
Comment (JEThompson97) [reply utils/clients/tests/test_queue_client.py:117]: Yeah, I see what you mean, I've started testing inner workings of job.Message (which are already covered by test_job).
Will resolve in next commit.

File: utils/clients/tests/test_queue_client.py (hunk 3 above)
Comment (JEThompson97) [reply utils/clients/tests/test_queue_client.py:105]: I'm a bit confused by this. `QueueClient.send` is what this test is testing, right? So by patching it out, wouldn't this test nothing except that "send is called when send is called"?

Comment (ElliotAOram) [reply utils/clients/tests/test_queue_client.py:105]: Yep you're right ignore me!

**Pull request merged**